### PR TITLE
End-to-end testing

### PR DESCRIPTION
**Pull request by benmor20**

### CODE REVIEW

File: .github/workflows/get-green-head.yml [hunk 1]
```diff
@@ -14,18 +14,25 @@ jobs:
...
+      - name: Run end-to-end tests in container
+        env:
+          APIKEY: ${{ secrets.GMAP_API_KEY }}
+        run: docker exec -w /code/testing/end_to_end_tests src-web-1 pytest --apikey "$APIKEY"
 
       - name: Set identity to Berwin
         run: git config --global user.email "blan@olin.edu" && git config --global user.name "Berwin Lan"
 
-      - name: Merge into main if tests pass
+      - name: Merge into green-head if tests pass
```
Comment (berwinlan) [inline .github/workflows/get-green-head.yml:32]: oop nice catch, thanks!

File: src/app.py [hunk 2]
```diff
@@ -45,14 +45,9 @@ def form():
         # Get keys, or return None if they don't exist
         start = request.form.get('start')
         end = request.form.get('end')
-        midpoints = request.form.get('midpoints')
+        midpoints = request.form.get('midpoints').split(',')
```
Comment (berwinlan) [inline src/app.py:48]: this .... this is such a better way of doing this yea idk why i did that earlier

File: testing/end_to_end_tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import pytest
```
Comment (berwinlan) [inline testing/end_to_end_tests/conftest.py:1]: consider changing file name to fit conventions (smth like `test_config`), but nbd

**Review by berwinlan — APPROVED** [review]
Looks great!

File: testing/end_to_end_tests/conftest.py (hunk 3 above)
Comment (benmor20) [reply testing/end_to_end_tests/conftest.py:1]: this isn't a testing file, afaik it's a special file `pytest` uses that (among other things) lets us pass command line arguments to the tests, hence why it's named weirdly

**Pull request merged**